htdocs/compta/stats/cabyprodserv.php

```diff
@@ -189,6 +196,16 @@
 	$sql = "SELECT DISTINCT p.rowid as rowid, p.ref as ref, p.label as label, p.fk_product_type as product_type,";
 	$sql.= " SUM(l.total_ht) as amount, SUM(l.total_ttc) as amount_ttc,";
 	$sql.= " SUM(CASE WHEN f.type = 2 THEN -l.qty ELSE l.qty END) as qty";
+	if($fk_supplier > 0) {
+		$sql.= ' ,(
+					SELECT SUM(cfd.qty) FROM '.MAIN_DB_PREFIX.'commande_fournisseur cf
+					INNER JOIN '.MAIN_DB_PREFIX.'commande_fournisseurdet cfd ON (cf.rowid = cfd.fk_commande)
+					WHERE cf.fk_soc = '.$fk_supplier.'
+					AND cf.date_commande >= "'.date('Y-m-d',$date_start).'" AND cf.date_commande <= "'.date('Y-m-d',$date_end).'"
+					AND cf.fk_statut >= 3
+					AND cfd.fk_product = p.rowid
+				  ) as qty_fourn';
+	}
```

Comment:
Having data on supplier order of an accounting turnover report per product looks strange.
Above all when it depends on a filter on a supplier. For a turnover report, no supplier should have effect on results.
currently, should be better to be moved into some dedicated reports on external modules.